components/script/dom/window.rs

```diff
@@ -584,6 +584,10 @@ impl WindowMethods for Window {
                     // Step 3, remove from the user interface
                     let _ = self.send_to_embedder(EmbedderMsg::CloseBrowser);
                     // Step 4, discard browsing context.
+                    // NOTE: Why is a call to discard_browsing_context neccessary to make
+                    // html/browsers/windows/auxiliary-browsing-contexts/opener-closed.html pass?
```

Comment:
IMO this question should have an answer before we land that. Cc @asajeffrey who worked on browsing contexts and whatnot.

Reply:
Ok, I'll debug a bit further...

Reply:
Ok, I was able to remove this, was actually a problem with how the auxiliary was created. 
The fix consisted of [creating a new joint-session-history for the new top-level to the ](https://github.com/servo/servo/pull/20678/commits/6e15fe931e189c7f7b89bca0fc184063136d233b#diff-55c92a6a5ba7654ce45fe6fc6c63740fR1712) and [creating the browsing context ID from the top-level one](https://github.com/servo/servo/pull/20678/commits/6e15fe931e189c7f7b89bca0fc184063136d233b#diff-b05366c9d370dbf97ed89b3cd7794d6eR218)